llvm-general/src/LLVM/General/OrcJIT.hs

```diff
@@ -12,7 +12,12 @@ module LLVM.General.OrcJIT (
     mangleSymbol,
     withIRCompileLayer,
     withModuleSet,
-    withObjectLinkingLayer
+    withObjectLinkingLayer,
+    addModuleSet,
+    createObjectLinkingLayer,
```

Comment:
I certainly agree `with` is not always sufficient, and examples like what you reference are the sort of discussion I'm looking for. There are many more complex approaches than the simple scoping allowed by `with`. It's just one of the simplest, and often sufficient. Leaving the problem up to the user of llvm-general, however, should be the option of last resort.

Can we hang the allocated objects off some long-lived object and clean them up when it goes out of scope, for instance? Are they interlinked, or can they be cleaned up in any order?